src/IoTSettings.ts

```diff
@@ -101,22 +81,22 @@ export class IoTWorkbenchSettings {
         if (showMessage) {
           throw new CancelOperationError('Change workbench cancelled.');
         }
-        return userWorkbenchPath;
+        return;
       }
     } else if (selection !== undefined) {
       userWorkbenchPath = selection.data;
     } else {
-      userWorkbenchPath = undefined;
+      userWorkbenchPath = '';

```

Comment:
I suggest to re-organize this function logic to make it clear, such as
1. show quick pick to ask user to select workbench folder.
2. if(!selection), return. (do we need to throw cancel error?)
3. if selection.data === '$', show open dialog for further selection.
4. else set the userWorkbenchPath value.
5. update config.